DEVELOPING.md

```diff
@@ -0,0 +1,199 @@
+# Developer guidelines
+
+This document contains information potentially relevant for GAP developers.
+Please also check out the file `CONTRIBUTING.md` for getting started working
+on GAP. In contrast, this file here contains more specialized information that
+you can use like a reference manual.
+
+
+## Branches
+
+At any time, there are two primary branches in the GAP git repository:
+
+- `master`: all new changes go into this; pull requests usually target it
+- `stable-4.X` (where `X` is some number): stable releases of the
+  `4.X.*` series are made from this branch
+
+We are using GitHub's "branch protection" feature to restrict direct pushes
+to these branches. All changes should go through pull requests, to ensure
+that tests are run automatically and are passing (hard experience taught us
+that even "trivial" and "obviously correct" changes can break things).
+
+Note that we only make releases from the most recent release branch; older
+release branches are kept around but we do not expect to e.g. make a new
+GAP 4.12.x release after 4.13.0 has been released, and so on.
+
+
+### Making changes to the release branch via backports
+
+By default all changes go into the `master` branch. Once a release 4.X.Y
+is out, to get changes into the 4.X.(Y+1) release, the usual process is
+to first land them into the `master` branch, and have them "backported"
+later on the.
+
+The process for backporting is as follows:
+1. someone adds the `backport-to-4.X` label to the pull request (if you
+   don't have permission, just ask for someone, e.g. @fingolfin, to add
+   the label)
+2. some time after the PR has been merged into `master`, and before the
+   next stable release, someone will go through the list of merged PRs
+   with this label, and backport them (usually by opening a dedicated
+   backporting PR, which may contain the content of multiple PRs being
+   backported).
+3. After a PR has been backported this way, the `backport-to-4.X` label
+   is replaced by a `backport-to-4.X-DONE` label, to avoid it from being
+   processed again (this is also important for our scripts which generate
```

Comment:
```suggestion
   is replaced by a `backport-to-4.X-DONE` label, to avoid
   processing it again (this is also important for our scripts which generate
```